source/common/unified_memory_allocation/include/uma/memory_provider.h

```diff
@@ -82,6 +82,45 @@ enum uma_result_t
 umaMemoryProviderGetLastResult(uma_memory_provider_handle_t hProvider,
                                const char **ppMessage);
 
+///
+/// \brief Retrieve an array of possible pages sizes used by memory region referenced by given ptr
+///        This array is sorted in ascending order.
+/// \param hProvider handle to the memory provider
+/// \param ptr pointer to memory allocated by this memory provider
+/// \param numPageSizes [in] number of memory providers to return. If numPageSizes is not equal to or
+///        greater than the number of possible page sizes, UMA_RESULT_ERROR_INVALID_ARGUMENT is returned.
+/// \param pageSizes [out] will be updated with page size values.
+/// \param numPageSizesRet [in] pointer to the actual number of possible page sizes
+/// \return UMA_RESULT_SUCCESS on success or appropriate error code on failure.
+enum uma_result_t
+umaMemoryProviderGetPageSizes(uma_memory_provider_handle_t hProvider, void *ptr,
+                              size_t numPageSizes, size_t *pageSizes,
+                              size_t *numPageSizesRet);
+
+///
+/// \brief Discard physical pages within the virtual memory mapping associated at given addr and size.
+///        This call is asynchronous and may delay puring the pages indefinitely.
+/// \param hProvider handle to the memory provider
+/// \param ptr beginning of the virtual memory range
+/// \param size size of the virtual memory range
+/// \return UMA_RESULT_SUCCESS on success or appropriate error code on failure.
```

Comment:
Should we consider that not all memory providers will support Purge functionality? I think we can add a corresponding return code to notify the caller of this API.